Lodgify.Payments.Stripe.Server.UnitTests/Controllers/v1/Externall/WebhookControllerTests.cs

```diff
@@ -0,0 +1,117 @@
+
+        // Assert
+        result.Should().BeAssignableTo<BadRequestObjectResult>();
+        var badRequestResult = (BadRequestObjectResult)result;
+        badRequestResult.Value.Should().Be("Stripe-Signature Header is missing");
+    }
+
+    [Fact]
+    public async Task HandleEvent_ShouldReturnOk_WhenEventIsHandledSuccessfully()
+    {
+        // Arrange
+        var payload = PayloadLoader.Load(PayloadType.AccountUpdatedChargesEnabledDetailsSubmitted);
+        var stripeSignature = StripeSignatureBuilder.GenerateSignature(payload, _stripeSettingsMock.Value.WebhookSecret);
+
+        var context = new DefaultHttpContext();
+        context.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(payload));
+        context.Request.Headers["Stripe-Signature"] = stripeSignature;
+        _controller.ControllerContext = new ControllerContext
+        {
+            HttpContext = context
+        };
+
+        // Act
+        var result = await _controller.HandleEvent(CancellationToken.None);
+
+        // Assert
+        result.Should().BeAssignableTo<OkResult>();
+        await _mediatorSenderMock.Received(1).Send(Arg.Any<AccountUpdatedCommand>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task HandleEvent_ShouldReturnBadRequest_WhenStripeExceptionIsThrown()
+    {
+        // Arrange
+        var payload = PayloadLoader.Load(PayloadType.AccountUpdatedStripeExceptionMaker);
+        var stripeSignature = StripeSignatureBuilder.GenerateSignature(payload, _stripeSettingsMock.Value.WebhookSecret);
+
+        var context = new DefaultHttpContext();
+        context.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(payload));
+        context.Request.Headers["Stripe-Signature"] = stripeSignature;
+        _controller.ControllerContext = new ControllerContext
+        {
+            HttpContext = context
+        };
+
+        // Act
+        var result = await _controller.HandleEvent(CancellationToken.None);
+
+
+        // Assert
+        result.Should().BeAssignableTo<BadRequestResult>();
+        await _mediatorSenderMock.DidNotReceive().Send(Arg.Any<AccountUpdatedCommand>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task HandleEvent_ShouldReturnBadRequest_WhenNotSupportedExceptionIsThrown()
+    {
+        // Arrange
+        var payload = PayloadLoader.Load(PayloadType.NotSupportedEventType);
+        var stripeSignature = StripeSignatureBuilder.GenerateSignature(payload, _stripeSettingsMock.Value.WebhookSecret);
+
+        var context = new DefaultHttpContext();
+        context.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(payload));
+        context.Request.Headers["Stripe-Signature"] = stripeSignature;
+        _controller.ControllerContext = new ControllerContext
+        {
+            HttpContext = context
+        };
+
+        // Act && Assert
+        await Assert.ThrowsAsync<NotSupportedException>(() => _controller.HandleEvent(CancellationToken.None));
```

Comment:
This also could be changed into fluent assertion.
I haven't tested the code but something like that should work:
```suggestion
        Func<Task> action = async () => { await _controller.HandleEvent(CancellationToken.None); }
       
       action.Should().ThrowAsync<NotSupportedException>();
```